WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderSearchSuggestionAdapter.java

```diff
@@ -1,47 +1,93 @@
 package org.wordpress.android.ui.reader.adapters;
 
 import android.content.Context;
-import android.database.MatrixCursor;
-import android.support.v4.widget.SimpleCursorAdapter;
+import android.database.Cursor;
+import android.support.v4.widget.CursorAdapter;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
 
+import org.wordpress.android.R;
 import org.wordpress.android.datasets.ReaderSearchTable;
 
-import java.util.List;
-
-public class ReaderSearchSuggestionAdapter extends SimpleCursorAdapter {
-    private List<String> mSearchSuggestions;
+public class ReaderSearchSuggestionAdapter extends CursorAdapter {
+    private static final int MAX_SUGGESTIONS = 5;
+    private String mCurrentFilter;
 
     public ReaderSearchSuggestionAdapter(Context context) {
-        super(context,
-                android.R.layout.simple_list_item_1,
-                null,
-                new String[]{"query"},
-                new int[]{android.R.id.text1},
-                0);
-        populate();
+        super(context, null, false);
     }
 
-    public void populate() {
-        populate(null);
+    /*
+     * populate the adapter using a cursor containing past searches that match the filter
+     */
+    public void setFilter(String filter) {
+        // skip if unchanged
+        if (isCurrentFilter(filter) && getCursor() != null) {
+            return;
+        }
+        Cursor cursor = ReaderSearchTable.getQueryStringCursor(filter, MAX_SUGGESTIONS);
+        swapCursor(cursor);
+        mCurrentFilter = filter;
     }
 
-    public void populate(String filter) {
-        mSearchSuggestions = ReaderSearchTable.getQueryStrings(filter);
-        MatrixCursor cursor = new MatrixCursor(new String[]{"_id", "query"});
+    /*
+     * forces setFilter() to always repopulate by skipping the isCurrentFilter() check
+     */
+    private void reload() {
+        String newFilter = mCurrentFilter;
+        mCurrentFilter = null;
+        setFilter(newFilter);
+    }
 
-        int id = 0;
-        for (String query : mSearchSuggestions) {
-            cursor.addRow(new Object[] {id++, query});
+    private boolean isCurrentFilter(String filter) {
+        if (TextUtils.isEmpty(filter) && TextUtils.isEmpty(mCurrentFilter)) {
+            return true;
         }
-
-        swapCursor(cursor);
+        return filter.equalsIgnoreCase(mCurrentFilter);
```

Comment:
`filter` could still be `null` when this line is reached - we should check it isn't null